syncmaster/worker/handlers/file/local_df.py

```diff
@@ -15,17 +16,27 @@
     from pyspark.sql import DataFrame
 
 
-class FileProtocolHandler(FileHandler):
+class LocalDFFileHandler(FileHandler):
 
     def read(self) -> DataFrame:
         from pyspark.sql.types import StructType
 
-        downloader = FileDownloader(
-            connection=self.file_connection,
-            source_path=self.transfer_dto.directory_path,
-            local_path=self.temp_dir.name,
-            filters=self._get_file_metadata_filters(),
-        )
+        downloader_params = {
+            "connection": self.file_connection,
+            "source_path": self.transfer_dto.directory_path,
+            "local_path": self.temp_dir.name,
+            "filters": self._get_file_metadata_filters(),
+        }
+
+        if self.transfer_dto.strategy.type == "incremental":
+            hwm_name = f"{self.transfer_dto.id}_{self.connection_dto.type}_{self.transfer_dto.directory_path}"
+            downloader_params["hwm"] = FileModifiedTimeHWM(
+                name=hwm_name,
+                expression=self.transfer_dto.strategy.increment_by,
+            )
```

Comment:
`increment_by` has values like `file_name`, `file_modified_time`, so it should be used to select HWM class, and not as `expression` field value. Actually, `FileDownloader` completely ignores `expression` value, it is important only for `DBReader`